### DISCUSSION TITLE
Airflow Upgrade Error - 2.2.2 to 2.2.4

### DISCUSSION THREAD

**Discussion by sridar-venkatesan** [apache/airflow#22500] [General]
Apache Airflow version
2.2.4 

What happened
We upgraded from Airflow 2.2.2 to 2.2.4 through pip. Upgrade was not successful as all our dags which had Python Operator broke with the following error:

  File "/home/airflow/airflowlibs/lib64/python3.7/site-packages/airflow/operators/python_operator.py", line 22, in <module>
    from airflow.operators.python import (  # noqa
  File "/home/airflow/airflowlibs/lib64/python3.7/site-packages/airflow/operators/python.py", line 36, in <module>
    from airflow.utils.operator_helpers import KeywordParameters
ImportError: cannot import name 'KeywordParameters' from 'airflow.utils.operator_helpers' (/home/airflow/airflowlibs/lib64/python3.7/site-packages/airflow/utils/operator_helpers.py)

What you think should happen instead
No response

How to reproduce
No response

Operating System
linux

Versions of Apache Airflow Providers
NA

Deployment
Other

Deployment details
PIP

Anything else
No response

Are you willing to submit PR?
No

[Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)
 I agree to follow this project's Code of Conduct

**Comment by potiuk** [comment]
It looks like some part of your system (likely workers) did not get the upgrade.

**Reply by potiuk** [reply]
Or simply you have not restarted them after the upgrade

**Reply by sridar-venkatesan** [reply]
tried restarting scheduler and webserver but its still the same. One difference is that the dag import errors number keeps changing everytime when i refresh. Any suggestions?

**Reply by potiuk** [reply]
I do not know what deployment you have and which executor you use, but in case you use Celery Executor, you need to restart workers.

**Reply by potiuk** [reply]
Also it might be that for some reason you have two airflow installed in your system (in different site packages or different virtualenvs) and sometimes the old version is imported first. That might mean that your installation is broken and I recommend to reinstall airflow from scratch (removing everything before doing it or generally starting everything - except the database) from scratch.

**Reply by sridar-venkatesan** [reply]
Its just a local executor. I think its resolved now, I saw many orphan threads of workers and schedulers. Cleared it and its working fine. 
Thanks for your help @potiuk
